nifi-nar-bundles/nifi-aws-bundle/nifi-aws-parameter-providers/src/main/java/org/apache/nifi/parameter/aws/AwsSecretsManagerParameterProvider.java

```diff
@@ -70,16 +70,38 @@
 @CapabilityDescription("Fetches parameters from AWS SecretsManager.  Each secret becomes a Parameter group, which can map to a Parameter Context, with " +
         "key/value pairs in the secret mapping to Parameters in the group.")
 public class AwsSecretsManagerParameterProvider extends AbstractParameterProvider implements VerifiableParameterProvider {
+    public static final String ENUMERATED_STRATEGY = "Enumerated";
+    public static final String PATTERN_STRATEGY = "Pattern";
+    public static final PropertyDescriptor SECRET_LISTING_STRATEGY = new PropertyDescriptor.Builder()
+            .name("secret-listing-strategy")
+            .displayName("Secret Listing Strategy")
+            .description("Strategy to use when listing secrets. 'Pattern' strategy uses Secret Name Pattern property and fetches  " +
+                    "all secrets whose names match the pattern. 'Pattern' strategy requires ListSecrets and GetSecretValue permissions. "+
+                    "'Enumerated' strategy uses 'Secret Names' property and fetches all secrets in the list. " +
+                    "'Enumerated' strategy requires only GetSecretValue permission.")
+            .required(true)
+            .allowableValues(PATTERN_STRATEGY, ENUMERATED_STRATEGY)
+            .defaultValue(PATTERN_STRATEGY)
+            .build();
 
     public static final PropertyDescriptor SECRET_NAME_PATTERN = new PropertyDescriptor.Builder()
             .name("secret-name-pattern")
             .displayName("Secret Name Pattern")
             .description("A Regular Expression matching on Secret Name that identifies Secrets whose parameters should be fetched. " +
-                    "Any secrets whose names do not match this pattern will not be fetched.")
+                    "Any secrets whose names do not match this pattern will not be fetched. " +
+                    "This parameter is ignored if Secret Listing Strategy parameter is set to 'Enumerated'.")
             .addValidator(StandardValidators.REGULAR_EXPRESSION_VALIDATOR)
-            .required(true)
+            .dependsOn(SECRET_LISTING_STRATEGY, PATTERN_STRATEGY)
             .defaultValue(".*")
             .build();
+
+    public static final PropertyDescriptor SECRET_NAMES = new PropertyDescriptor.Builder()
+            .name("secret-names")
+            .displayName("Secret Names")
+            .description("Comma-separated list of secret names to fetch. This parameter is ignored if the Secret Listing Strategy parameter is set to 'Pattern'.")
```

Comment:
As outlined before, no need to document this due to `dependsOn`.

```suggestion
            .description("Comma-separated list of secret names to fetch.")
```